include/model.php

```diff
@@ -47,35 +67,38 @@ class PLL_Model {
 	public $term;
 
 	/**
-	 * Flag set to true during the language objects creation.
+	 * Model for the languages.
 	 *
-	 * @var bool
+	 * @var Models\Language_Model
 	 */
-	private $is_creating_language_objects = false;
+	public $language_model;
 
 	/**
-	 * Tells if {@see PLL_Model::get_languages_list()} can be used.
+	 * Model for taxonomies filtered by Polylang.
 	 *
-	 * @var bool
+	 * @var Models\Filtered_Taxonomies_Model
 	 */
-	private $languages_ready = false;
+	public $filtered_taxonomies_model;
 
 	/**
 	 * Constructor.
 	 * Setups translated objects sub models.
 	 * Setups filters and actions.
 	 *
 	 * @since 1.2
+	 * @since 3.7 Type of parameter `$options` changed from `array` to `Options`.
 	 *
-	 * @param array $options Polylang options.
+	 * @param Options $options Polylang options.
 	 */
-	public function __construct( &$options ) {
-		$this->options = &$options;
+	public function __construct( Options &$options ) {
+		$this->options                   = &$options;
+		$this->cache                     = new PLL_Cache();
```

Comment:
Could we have a cache per object instead of a shared one?
i.e. one for `Language_Model`, one for `PLL_Translated_Post` and another `PLL_Translated_Term`.

Reply:
The same object was used everywhere so far, any reason to separate everything?

Reply:
The purpose is to remove `PLL_Model::$cache` property which is useless to this very class and only useful others.

Reply:
`PLL_Model::$cache` is used in `PLL_Model::count_posts()` :/

Reply:
Damn...